packages/melos/lib/src/package.dart

```diff
@@ -531,54 +531,52 @@ class PackageMap {
 
     final packageMap = <String, Package>{};
 
-    await Future.wait<void>(
-      pubspecFiles.map((pubspecFile) async {
-        final pubspecDirPath = pubspecFile.parent.path;
-        final pubspec = Pubspec.parse(pubspecFile.readAsStringSync());
-        final name = pubspec.name;
-
-        if (packageMap.containsKey(name)) {
-          throw MelosConfigException(
-            '''
+    await Stream.fromIterable(pubspecFiles).parallel((pubspecFile) async {
```

Comment:
I just realized that we should have a higher number here than the number of cores, since this isn't running on isolates. I don't know what a fitting number would be though, maybe try a few different and see how it affects performance?

Reply:
For context, the project has 294 packages. I've run `time melos list` three times for each case below (default, 25, 50, and 100) and didn't see much improvement. The higher numbers will increase the risk of "too many open files" issue

Here are some benchmarks when `parallelism`:
- was null (number of cores by default which is 10 on my device):
```
melos list  1.27s user 0.68s system 89% cpu 2.179 total
melos list  1.31s user 0.71s system 92% cpu 2.164 total
melos list  1.32s user 0.77s system 90% cpu 2.311 total
```
- was 25:
```
melos list  1.32s user 0.72s system 68% cpu 2.968 total
melos list  1.31s user 0.74s system 89% cpu 2.285 total
melos list  1.28s user 0.72s system 91% cpu 2.204 total
```
- was 50:
```
melos list  1.32s user 0.72s system 88% cpu 2.312 total
melos list  1.30s user 0.74s system 86% cpu 2.351 total
melos list  1.31s user 0.75s system 88% cpu 2.341 total
```
- was 100:
```
melos list  1.32s user 0.74s system 86% cpu 2.375 total
melos list  1.30s user 0.73s system 91% cpu 2.220 total
melos list  1.31s user 0.73s system 89% cpu 2.290 total
```

Reply:
Just out of curiosity, what does it say with 1?

Reply:
The first three run:
```
melos list  1.27s user 0.75s system 54% cpu 3.752 total
melos list  1.26s user 0.70s system 90% cpu 2.161 total
melos list  1.29s user 0.72s system 90% cpu 2.212 total
```

Interesting :)

Reply:
Seems like we don't need parallelism then 😄

Reply:
I'm sorry for late response. I was lazy these days. You're right, there is no need for parallelism. Just noticed that pubspec.yaml files are read sync

Reply:
No worries! Can you compare the performance with this PR and prior to this PR for a large repository, but not large enough so that it crashes crash on your OS?